Sources/UI/Chat/Composer/ComposerView.swift

```diff
@@ -273,11 +273,11 @@ public extension ComposerView {
     }
     
     internal func updateSendButton() {
-        let isAnyFileUploaded = uploadManager?.images.first(where: { $0.attachment != nil }) != nil
-            || uploadManager?.files.first(where: { $0.attachment != nil }) != nil
+        let anyUploadedItem = uploadManager?.images.first(where: { $0.attachment != nil })
+            ?? uploadManager?.files.first(where: { $0.attachment != nil })
         
         if let style = style {
-            let isHidden = text.isEmpty && !isAnyFileUploaded
+            let isHidden = text.isEmpty && anyUploadedItem == nil
```

Comment:
What about something like this:
```
        var hasUploadedItem = false
        if let uploadManager = uploadManager {
            let hasUploadedImages = uploadManager.images.contains(where: { $0.attachment != nil })
            let hasUploadedFiles = uploadManager.files.contains(where: { $0.attachment != nil })
            hasUploadedItem = hasUploadedImages || hasUploadedFiles
        }

        if let style = style {
            let isHidden = text.isEmpty && hasUploadedItem
            
```

Reply:
I think in my case is a pattern to find any uploaded item in 2 lines and the second line is optional if the first one will find an item.
In your case, 6 lines and all of them will be executed.
As a compromise, I think I can add a comment for easier reading.

Reply:
I don't think we're in the fewer-lines-win competition here. What's easier to read and understand is what matters. @b-onc we need a third opinion here.

Reply:
Not because it's fewer lines, but because second expression doesn't need to be run, and I didn't feel like it's hard to read, I'd suggest current way with a minor change

```swift
guard let uploadManager = uploadManager else {
  // no need to check for style of text
  sendButton.isHidden = false
  return
}

let hasUploadedItem = uploadManager.contains(where: {...}) || uploadManager.contains(where: {...})
```

or what @VojtaStavik suggested without `hasUploadedImages` and `hasUploadedFiles` variables (so || expressions directly)

Reply:
@b-onc 
- like the `guard` in the beginning 👍 
- with writing it in one expression you're just doing compiler's job. in my original code, the compiler optimizes it such that if `hasUploadedImages` was true,  `hasUploadedFiles` would never be evaluated, too. Let the compiler do its job. Your job is to write it so your teammates don't have to think about it ;-) 
- we're talking about optimizing code which runs just once and has no impact unless there will be millions of items there (not possible)

Reply:
@VojtaStavik Are you sure the compiler optimizes the code? When you write the expressions explicitly it doesn't optimize them, do you have any resources so I can read up on? I've tried:
```swift
func one() -> Bool {
    let res = Bool.random()
    print("one: \(res)")
    return res
}

func two() -> Bool {
    let res = Bool.random()
    print("two: \(res)")
    return res
}

let oneRes = one()
let twoRes = two()

let val = oneRes || twoRes

if val {
    print("val true")
} else {
    print("val false")
}
```
and when `one()` returns `true`, `two()` is still executed.

If that's the case and compiler optimizes the code, totally agreed that we can have them explicit.

Performance is one thing, and while I do agree that our gains are very minimal here, I don't think readability is taking a hit. I feel writing two expressions explicitly is a bit -unnecessarily- more verbose.

Reply:
You know what, you're probably right! I didn't think about the side effects. Sorry. I will be more careful about my statements.

@buh Pick the solution you like the most and let's merge this!

Reply:
First of all, I like this conversation because we can figure each other.
1. Yes, I tried to avoid the side effect.
2. I don't think we need a guard for `uploadManager`. It's because we no need it if the user wants to send a message. Some of our customers disable all attachments (basically removed ⊕) and we can optimize that and skip setting upload manager for `ComposerView`, so it will be always `nil`. But I added a guard for the style.
3. I don't want to rename `anyUploadedItem` to `hasUploadedItem` because it will increase complexity as it was at the beginning in one line:
```swift
let x = (a... ?? b...) != nil
```

Can I merge now?

Reply:
OK 👍